pkg/api/api.go

```diff
@@ -110,17 +110,19 @@ func Router() *gin.Engine {
 				licenses.GET("/preview", GetAllLicensePreviews)
 				licenses.POST("", CreateLicense)
 				licenses.PATCH(":shortname", UpdateLicense)
-				licenses.POST("import", ImportLicenses)
+				licenses.POST("import", middleware.RoleBasedAccessMiddleware([]string{"ADMIN"}), ImportLicenses)
 			}
 			search := authorizedv1.Group("/search")
 			{
 				search.POST("", SearchInLicense)
 			}
 			users := authorizedv1.Group("/users")
 			{
-				users.GET("", auth.GetAllUser)
-				users.GET(":id", auth.GetUser)
+				users.GET("", middleware.RoleBasedAccessMiddleware([]string{"ADMIN"}), auth.GetAllUser)
+				users.GET(":username", middleware.RoleBasedAccessMiddleware([]string{"ADMIN"}), auth.GetUser)
 				users.POST("", auth.CreateUser)
+				users.PATCH(":username", middleware.RoleBasedAccessMiddleware([]string{"ADMIN"}), auth.UpdateUser)
```

Comment:
This will prevent user from modifying their own passwords.

Better allow everyone to hit this endpoint, but allow only admins to update:
* Fields for different user
* Role for anyone
* Password and email only for current user.

Reply:
Alternatively, let the `PATCH /users/:username` be accessible only for admins. Create a new `PATCH /users` for non-admin users to modify their own fields (other than username and email).